pkg/urbit/noun/retrieve.c

```diff
@@ -1107,6 +1107,39 @@ u3r_mp(mpz_t   a_mp,
   }
 }
 
+/* u3r_short():
+**
+**   Return short (a_w) of (b).
+*/
+c3_s
+u3r_short(c3_w  a_w,
+          u3_atom b)
+{
+  c3_assert( u3_none != b );
+  c3_assert( c3y == u3a_is_atom(b) );
+
+  if ( c3y == u3a_is_cat(b) ) {
+    switch ( a_w ) {
+      case 0:  return b & 0xffff;
+      case 1:  return b >> 16;
```

Comment:
I'd expect `0` to be no data at all, and `1` to be two bytes (which is what a short is, right?)... so I'm probably misunderstanding what this is supposed to be doing.

Reply:
`a_w` is a zero-based index into the atom, considered as an array of shorts (ie, uint16). In the case of a direct atom, `u3r_short(0, a)` is the 16 least-significant bits, and `u3r_short(1, a)` is the 16 most significant bits.

This is the same interface as `u3r_bytes()` and `u3r_words()`, modulo the difference in size.

Reply:
> his is the same interface as `u3r_bytes()` and `u3r_words()`

`_byte` and `_word`, right?

This makes sense now, in any case. Thanks!

Reply:
Right, that's what I meant.

Reply:
I thought about adding `u3r_shorts()` as well, but I don't have an immediate use-case, and it raises (new) endian-ness questions.